functions/webhooks/instagram/InstagramToFlex.ts

```diff
@@ -59,18 +59,19 @@ export type Body = InstagramMessageEvent & {
  */
 const isValidFacebookPayload = (event: Body, appSecret: string) => {
   if (!event.bodyAsString || !event.xHubSignature) return false;
-
-  const expectedSignature = crypto
-    .createHmac('sha1', appSecret)
-    .update(event.bodyAsString)
-    .digest('hex');
-
-  const isValidRequest = crypto.timingSafeEqual(
-    Buffer.from(event.xHubSignature),
-    Buffer.from(`sha1=${expectedSignature}`),
-  );
-
-  return isValidRequest;
+  try {
+    const expectedSignature = crypto
+      .createHmac('sha1', appSecret)
+      .update(event.bodyAsString)
+      .digest('hex');
+
+    return crypto.timingSafeEqual(
+      Buffer.from(event.xHubSignature),
+      Buffer.from(`sha1=${expectedSignature}`),
+    );
+  } catch (e) {
+    return false;
+  }
```

Comment:
If you want to throw 500 that's ok, but the code as was was throwing an exception that wasn't resolving to any status code, it was just unhandled, and the does function throw errors for client issues, like the signature being the wrong length

My only concern about changing it from a 4xx to a 5xx is that this code is before we've even established the request is from FB, so it could be open to deliberate or accidental abuse. But we can put the original function inside the below try block if you prefer. I'll put a console.warn in for now but can amend further if you prefer - I wasn't sure if we should be logging